### ISSUE TITLE
[openmp] Some libarcher tests fail intermittently

### ISSUE THREAD

**Issue by tuliom** [llvm/llvm-project#127796]
I've been seeing this issue happen from time to time.
So far, I've reproduced this issue with:
- `libarcher :: races/task-two.c`
- `libarcher :: races/lock-nested-unrelated.c`

The error message is usually like this one:

```
# | /builddir/build/BUILD/llvm-21.0.0_pre20250217.g26fc2e90fc5431-build/llvm-project-26fc2e90fc54313bfe3e0e1fbbb14251ed8afe29/openmp/tools/archer/tests/races/task-two.c:44:11: error: CHECK: expected string not found in input
# | // CHECK: ThreadSanitizer: reported {{[0-9]+}} warnings
# |           ^
# | <stdin>:29:5: note: scanning from here
# | DONE
# |     ^
# | <stdin>:30:1: note: possible intended match here
# | ThreadSanitizer: thread T4 finished with ignores enabled, created at:
# | ^
```

As it can be seen, the test is more or less prepared to deal with the fact that tsan is not 100% accurate in detecting and reporting all data races.
I suspect the problem is: when no data races is detected, that line is not printed and these tests fail.
